src/core/public/fatal_errors/fatal_errors_screen.tsx

```diff
@@ -0,0 +1,145 @@
+
+import { ErrorInfo } from './get_error_info';
+
+interface Props {
+  kibanaVersion: string;
+  buildNumber: number;
+  errorInfo$: Rx.Observable<ErrorInfo>;
+}
+
+interface State {
+  errors: ErrorInfo[];
+}
+
+export class FatalErrorsScreen extends React.Component<Props, State> {
+  public state: State = {
+    errors: [],
+  };
+
+  private subscription?: Rx.Subscription;
+
+  public componentDidMount() {
+    // reload the page if hash-based navigation is attempted
+    window.addEventListener('hashchange', () => {
+      window.location.reload();
+    });
+
+    // consume error notifications and set them to the component state
+    this.subscription = this.props.errorInfo$
+      .pipe(
+        tap(error => {
+          this.setState(state => ({
+            ...state,
+            errors: [...state.errors, error],
+          }));
+        })
+      )
+      .subscribe({
+        error(error) {
+          // tslint:disable-next-line no-console
+          console.error('Uncaught error in fatal error screen internals', error);
+        },
+      });
+  }
+
+  public componentWillUnmount() {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+      this.subscription = undefined;
+    }
+  }
+
+  public render() {
+    return (
+      <EuiPage style={{ minHeight: '100vh' }}>
+        <EuiPageBody>
+          <EuiPageContent verticalPosition="center" horizontalPosition="center">
+            <EuiEmptyPrompt
+              iconType="alert"
+              iconColor="danger"
+              title={<h2>Something went wrong</h2>}
+              body={
+                <p>
+                  Try refreshing the page. If that doesn't work, go back to the previous page or
+                  clear your session data.
+                </p>
+              }
+              actions={[
+                <EuiButton
+                  color="primary"
+                  fill
+                  onClick={this.onClickClearSession}
+                  data-test-subj="clearSession"
+                >
+                  Clear your session
+                </EuiButton>,
+                <EuiButtonEmpty onClick={this.onClickGoBack} data-test-subj="goBack">
+                  Go back
+                </EuiButtonEmpty>,
+              ]}
+            />
+            {this.state.errors.map((error, i) => (
+              <EuiCallOut key={i} title={error.message} color="danger" iconType="alert">
+                <EuiCodeBlock language="bash" className="eui-textBreakAll">
+                  {`Version: ${this.props.kibanaVersion}` +
+                    '\n' +
+                    `Build: ${this.props.buildNumber}` +
+                    '\n' +
+                    (error.stack ? error.stack : '')}
+                </EuiCodeBlock>
+              </EuiCallOut>
+            ))}
+          </EuiPageContent>
+        </EuiPageBody>
+      </EuiPage>
+    );
+  }
+
+  private onClickGoBack = (e: React.MouseEvent<HTMLButtonElement>) => {
```

Comment:
__note:__ that's funny, in Kibana we use both `onSomethingClick` and `onClickSomething` naming styles :)